File: backend/project/main/kotlin/challenge/domain/payment/Payment.kt
package challenge.domain.payment

import challenge.domain.order.Invoice
import challenge.domain.order.Order
import java.util.*

data class Payment(val order: Order, val paymentMethod: PaymentMethod) {
    val paidAt = Date()
    val authorizationNumber = paidAt.time
    val amount = order.totalAmount
    val invoice = Invoice(order)
}

File: backend/individual-assignment/src/models/Payment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Payment = void 0;
const Invoice_1 = require("./Invoice");
class Payment {
    constructor(order, paymentMethod) {
        this.order = order;
        this.paymentMethod = paymentMethod;
        this.paidAt = new Date();
        this.authorizationNumber = this.paidAt.getTime();
        this.amount = order.totalAmount;
        this.invoice = new Invoice_1.Invoice(order);
    }
}
exports.Payment = Payment;
